odoo/service/security.py

```diff
@@ -24,5 +17,4 @@ def check_session(session, env):
     expected = self._compute_session_token(session.sid)
     if expected and odoo.tools.misc.consteq(expected, session.session_token):
         return True
-    self._invalidate_session_cache()
```

Comment:
As discussed on a previous version of the commit, at that point in the `check_session()` the cache is supposed to be valid, there is no reason to invalidate it, so we can remove this invalidation.
It only happened when there was a mismatch, so the session had a UID but the token did not match the expected token.
- Either we just had a cache miss for the token, so we'd just refreshed it with the new value, nothing to invalidate
- Or we had a cache hit, so there's perhaps another session for the same user that has seeded the cache. But the cached valued must be good because we invalidate and propagate the invalidation whenever any user attribute changes (cfr `res.users.write()`)